src/ControlPoint.cpp

```diff
@@ -110,11 +110,13 @@ void ControlPoint::processDescriptionResponse(HttpConnection& connection, Descri
 
 bool ControlPoint::sendRequest(HttpRequest* request)
 {
-	// Don't create reponse stream until headers are in: this allows requests to be queued
-	request->onHeadersComplete([this](HttpConnection& client, HttpResponse& response) -> int {
-		client.getRequest()->setResponseStream(new LimitedMemoryStream(maxDescriptionSize));
-		return 0;
-	});
+	// Don't create response stream until headers are in: this allows requests to be queued
+	if(request != nullptr && request->getResponseStream() == nullptr) {
```

Comment:
Good point, agreed. However, things could get messy if `sendDescriptionRequest` is called with a response stream already set. This could be managed by adding `setResponseStream(nullptr);` at line 124, unless you have any better idea?

Reply:
> Good point, agreed. However, things could get messy if sendDescriptionRequest is called with a response stream already set. 

Why? Every request's response stream should receive its data as expected. Actually setting it at the start or onHeadersComplete should not matter. If that is not the case then the HttpClient should be fixed. 

Can you give me an example and steps to reproduce the issue that you have experienced?

Reply:
Well, `sendDescriptionRequest` sets a callback which expects a memory stream. If the caller sets some other stream type it'll go pop.

Reply:
NB. I also don't set the response stream until later because if there are multiple requests we'll be out of memory very quickly. The intent is to have only one description stream active at any time.

Reply:
Actually, at present it seems that the only call to `sendDescriptionRequest` is within `ControlPoint::requestDescription`, so perhaps either merge the code into that method or make it private.